Modules/_xxsubinterpretersmodule.c

```diff
@@ -347,12 +347,165 @@ channel_exceptions_init(PyObject *ns)
     return 0;
 }
 
+typedef struct _channelitem_wait_lock {
+    PyThread_type_lock lock;
+    PyThread_type_lock mutex;
+    int lock_is_allocated;
+    int is_sent;
+    int is_recv;
+} _channelitem_wait_lock;
+
+/**
+ * Allocate a new wait lock for a channel item.
+ *
+ * This function allocates memory that must be freed by calling
+ * _channelitem_wait_lock_recv and _channelitem_wait_lock_wait.
+ */
+static void *
+_channelitem_wait_lock_new(void)
+{
+    _channelitem_wait_lock *wait_lock = calloc(1, sizeof(*wait_lock));
+    wait_lock->lock = PyThread_allocate_lock();
+    wait_lock->mutex = PyThread_allocate_lock();
```

Comment:
It looks like `mutex` protects the struct's members.  Is that right?  If so, do we really need it?  There are only 2 threads that may be touching the state: the one where `send()` was called in the original interpreter and the one where `recv()` is called.  I didn't see anywhere there could be a race such that you need that mutex.